File: test_regress/t/t_display_merge.pl
#!/usr/bin/perl
if (!$::Driver) { use FindBin; exec("$FindBin::Bin/bootstrap.pl", @ARGV, $0); die; }
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# Copyright 2003 by Wilson Snyder. This program is free software; you
# can redistribute it and/or modify it under the terms of either the GNU
# Lesser General Public License Version 3 or the Perl Artistic License
# Version 2.0.
# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0

scenarios(simulator => 1);

compile(
    verilator_flags2=>["--stats"],
    );

execute(
    check_finished => 1,
    expect_filename => $Self->{golden_filename},
    );

file_grep("$Self->{obj_dir}/$Self->{VM_PREFIX}__stats.txt",
          qr/Node count, DISPLAY \s+ 32 \s+ 25 \s+ 25 \s+ 25/);

ok(1);
1;

File: test_regress/t/t_display_merge.py
#!/usr/bin/env python3
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# Copyright 2024 by Wilson Snyder. This program is free software; you
# can redistribute it and/or modify it under the terms of either the GNU
# Lesser General Public License Version 3 or the Perl Artistic License
# Version 2.0.
# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0

import vltest_bootstrap

test.scenarios('simulator')

test.compile(verilator_flags2=["--stats"])

test.execute(expect_filename=test.golden_filename)

test.file_grep(test.obj_dir + "/" + test.vm_prefix + "__stats.txt",
               r'Node count, DISPLAY \s+ 44 \s+ 27 \s+ 27 \s+ 6')

test.passes()
